### ISSUE TITLE
Execute each campaign in own environment

### ISSUE THREAD

**Issue by akorb** [secdev/scapy#2535]
### Problem
The current test approach adds a dependency on the order of the test execution.
It might succeed in one test execution order but fail in another.
It might succeed locally but fail on the Travis server.
I think Unit Tests should be independent units which are reliable and not rely on side effects.
For clarification, I only talk about the order of campaigns/uts files, not about the containing test sets or tests.

### Example 1
Some automotive related tests import _ISOTPSocket_ what, depending on the execution environment, actually imports _ISOTPSoftSocket_ or _ISOTPNativeSocket_. Now if _ISOTPSoftSocket_ is imported in one test and _ISOTPNativeSocket_ should have been imported in a later one through configuration, _ISOTPSoftSocket_ is still used since it was already imported and python does not reimport.

### Example 2
Execution Order 1:
```C
myTests1.uts // succeeds
myTests2.uts // succeeds
```

Execution Order 2:
```C
myTests2.uts // succeeds
myTests1.uts // fails
```

### Idea
Execute each campaign in an own python process and get its output over _stdout_. I guess it shouldn't be very time costly and would add a lot of reliability to the unit tests.
This would also allow running tests in parallel.

**Comment by gpotter2** [comment]
This used to be the case, but loading scapy over and over doubles the load time... when we had fewer files, it already took for ever, so nowadays it would be even worse...
see https://github.com/secdev/scapy/pull/543

**Comment by guedou** [comment]
I wonder if moving from UTScapy to pytest will improve this situation. I have being thinking of porting the unit tests to pytest in order to benefit from tests parallelization.

**Comment by guedou** [comment]
I have a silly idea. Could we use the https://docs.python.org/3/library/functions.html?highlight=globals#globals builtin function to achieve a better reliability ?
For example, backuping and restoring at https://github.com/secdev/scapy/blob/master/scapy/tools/UTscapy.py#L1057 ?

**Comment by polybassa** [comment]
I made a PoC for a transfer to pytest in #2550

**Comment by polybassa** [comment]
This issue should be fixed with #2545
